### ISSUE TITLE
gatsby serve Command failed with signal "SIGSEGV" in rootless docker image 

### ISSUE THREAD

**Issue by paulsouche** [gatsbyjs/gatsby#38108]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Hello 

The command `gatsby serve --port 8080 --host 0.0.0.0` fails with SIGSEGV in a docker container when the user is not root.

I checked all the [related issues](https://github.com/gatsbyjs/gatsby/issues?q=is%3Aissue+SIGSEGV+is%3Aclosed) and [one in particular](https://github.com/gatsbyjs/gatsby/issues/37246#issuecomment-1350791564) which seemed to be fixed in v5.3.1.

The project was using [gatsby@^4.25.1](https://github.com/SocialGouv/recosante/blob/f25633f345ab18b0d9917c141ff5306828c9d0a8/frontend/yarn.lock#L6358) so I did a PR to bump [gatsby@^5.10.0](https://github.com/SocialGouv/recosante/blob/master/yarn.lock#L9899) but even with that, the command is still failing.

Thanks anyway for the great job

### Reproduction Link

https://github.com/SocialGouv/recosante

### Steps to Reproduce

1. Clone the repo
2. Update the [frontend Dockerfile](https://github.com/SocialGouv/recosante/blob/master/frontend/Dockerfile) to use the `nodejs:frontend` user
3. In root folder run `docker-compose build frontend`
4. Run the container => exit code 139 SIGSEGV
...

### Expected Result

The command should succeed even with a non root user

### Actual Result

The command fails with a non root user

### Environment

```shell
System:
    OS: Linux 5.19 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
    CPU: (8) x64 Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 18.16.0 - /tmp/yarn--1685005231996-0.33751789578990987/node
    Yarn: 1.22.19 - /tmp/yarn--1685005231996-0.33751789578990987/yarn
    npm: 9.5.1 - /usr/local/bin/npm
  Languages:
    Python: 3.10.6 - /usr/bin/python
  Browsers:
    Chrome: 113.0.5672.92
    Firefox: 113.0.1
```

### Config Flags

NA

**Comment by suresh-jbt** [comment]
Yes, Having the same issue with me as well with the latest version.
I'm using a Mac. It worked once after reinstalling node_modules but 2nd time it had throw same error.

**Comment by carlqt** [comment]
> Yes, Having the same issue with me as well with the latest version. I'm using a Mac. It worked once after reinstalling node_modules but 2nd time it had throw same error.

My issue is closer to this. I'm not running it in a container.

As a workaround, I have to run `gatsby clean` before running `gatsby develop`
